### ISSUE TITLE
PDFDocumentProxy : destroy() doesn't work :  memory bloat

### ISSUE THREAD

**Issue by ineedmny** [VadimDez/ng2-pdf-viewer#414]
##### Bug Report or Feature Request (mark with an `x`)
```
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [X] Bug report -> please search issues before submitting
- [ ] Feature request
- [ ] Documentation issue or request
```

```
Ionic:

   ionic (Ionic CLI)  : 4.3.1 (/usr/local/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.0

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.2.5, (and 6 other plugins)

System:

   ios-deploy : 1.9.2
   ios-sim    : 6.0.0
   NodeJS     : v11.2.0 (/usr/local/bin/node)
   npm        : 6.4.1
   OS         : macOS
   Xcode      : Xcode 10.1 Build version 10B61
```

Hi,

i wondered why my Ionic app stuck after exactly 9 times loading a pdf.

After some debugging in Chrome DevTools i found out, that each Document stays in the memory and makes the app increasingly slower.

On component destroy the Pdf is getting destroyed.

```typescript
import { PDFDocumentProxy } from 'ng2-pdf-viewer';

@Component({
  selector: 'pdf-viewer',
  templateUrl: 'pdf-viewer.html'
})
export class PdfViewerComponent {

    public loadedPdf: PDFDocumentProxy;

    constructor() { }

    afterLoadComplete(pdf: PDFDocumentProxy) {
        this.loadedPdf = pdf;
    }

    ngOnDestroy() {
        this.loadedPdf.destroy();
    }

}
```

```html
<pdf-viewer
    [src]="pdf"
    [render-text]="false"
    [fit-to-page]="true"
    (after-load-complete)="afterLoadComplete($event)"
    (page-rendered)="pageRendered($event)"
    (error)="onError($event)"
    style="display: block;">
</pdf-viewer>
```

I don't know what i can do next. I tried some things, but the app still gets slower.
Thank you for your help!

**Comment by strerene** [comment]
Hey, 
we are experiencing the same issue.

After opening and clsing a document 6 times, the PDFViewer stays in memory exactly 6 times. This makes the library unusable on iOS since it starts displaying only blank pages after a while and decreases the performance of the whole application drastically.

![image](https://user-images.githubusercontent.com/41798235/55956117-7ff82f80-5c63-11e9-8f93-0f0dde29b0c4.png)

**Comment by jitendrak414** [comment]
Hey 
we are facing the same issue, 
I am really thankful if anyone can help me out.

Thanks

**Comment by vishnu-dev** [comment]
@VadimDez or anyone, Do you have any idea on how to solve this?

**Comment by VadimDez** [comment]
Fixed in `6.0.0` See https://github.com/VadimDez/ng2-pdf-viewer/blob/master/CHANGELOG.md#600
